### PR TITLE
Added support for specifying number of decimals

### PR DESCRIPTION
**Pull request by rowanwins**
Gday @nerik 

I've got a use case where I dont really want to show all the decimal places (particular with the seconds) so I thought'd I'd see if you're interested in adding a parameter to specify the precision of decimal places.

What do you think?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,7 +37,7 @@ var coords = formatcoords(27.725499,-18.024301);
 ```
 
 ```
-coords.format([format, lat/lon separator])
+coords.format([format, lat/lon separator, number of decimal places])
```
Comment (nerik) [inline README.md]: I think at this point we should switch to a different method signature:

```
coords.format([format], [params])
```

Where format and params are optional, and params is an object with optional parameters: 

```
{
   latLonSeparator (string, defaults to ' ')
   decimalPlaces (int, defaults to 5)
}
```

Following cases should be handled (also in tests and doc)
- defaults values when `params` is not specified and when keys inside `params` are not specified
- case when `format` is omitted (type detection)
- case when second parameter is a string, in which case assume that it's lat/lon spearator (for retro compatibility)

File: index.js [hunk 2]
```diff
@@ -67,29 +67,33 @@ var units = {
...
 		formatted = formatted.replace(/MM/g, values.minutesInt+units.minutes);
-		formatted = formatted.replace(/mm/g, values.minutes.toFixed(5)+units.minutes);
+		formatted = formatted.replace(/mm/g, values.minutes.toFixed(numberDecimals)+units.minutes);
 		formatted = formatted.replace(/M/g, values.minutesInt);
-		formatted = formatted.replace(/m/g, values.minutes.toFixed(5));
-		formatted = formatted.replace(/ss/g, values.seconds.toFixed(5)+units.seconds);
-		formatted = formatted.replace(/s/g, values.seconds.toFixed(5));
+		formatted = formatted.replace(/m/g, values.minutes.toFixed(numberDecimals));
+		formatted = formatted.replace(/ss/g, values.seconds.toFixed(numberDecimals)+units.seconds);
+		formatted = formatted.replace(/s/g, values.seconds.toFixed(numberDecimals));
```
Comment (nerik) [inline index.js:118]: This section could be more DRY, but original code is to blame. Can be fixed in a later code style PR.

**Comment by nerik** [comment]
Hi Rowan. This is a great addition, thanks.
I'd like to keep function signatures short, see [comment](https://github.com/nerik/formatcoords/pull/3#discussion_r65008966)
Let me know what you think and if you're interested in making the edits I suggested :)

**Comment by rowanwins** [comment]
Gday @nerik 
Yep you're suggestions make sense. I'll probably wont get time to incorporate all the changes for a few days so if you get want to get things done sooner feel free to take a lead otherwise I'll get around to it when I have a spare evening.

Cheers
Rowan

**Comment by rowanwins** [comment]
Gday @nerik 

I've pushed a new commit as per your suggetsions. Have a look over things and hopefully you're happy with them. Have included a few more tests and they all seem to be working.

Cheers
Rowan

**Comment by nerik** [comment]
Looks good, thanks 👍

**Pull request merged**